apps/services/user-profile/src/app/user-profile/userProfile.controller.ts

```diff
@@ -153,12 +220,20 @@ export class UserProfileController {
     allowEmptyValue: false,
   })
   @ApiCreatedResponse({ type: EmailVerification })
+  @Audit<EmailVerification>({
+    resources: (emailVerification) => emailVerification?.nationalId ?? '',
+  })
   async recreateVerification(
-    @Param('nationalId', UserProfileByNationalIdPipe)
-    profile: UserProfile,
-  ): Promise<EmailVerification | null> {
+    @Param('nationalId')
+    nationalId: string,
+    @CurrentUser()
+    user: User,
+  ): Promise<EmailVerification> {
+    // findOneByNationalId must be first as it implictly checks if the
+    // route param matches the authenticated user.
+    const profile = await this.findOneByNationalId(nationalId, user)
     if (!profile.email) {
-      return null
+      throw new BadRequestException('Profile have no configured email address.')
```

Comment:
'Profile has no ...' or '... does not have a configured ...'

Reply:
Takk!